mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstractseparecurring.php

```diff
@@ -14,9 +14,7 @@ public function __construct ()
     {
         parent::__construct();
         $directDebit = new Mollie_WC_Gateway_DirectDebit();
```

Comment:
Deze class `Mollie_WC_Gateway_AbstractSubscription` wordt door verschillende andere classes overschreven. 

Allen komen uiteindelijk terecht in de `Mollie_WC_Gateway_Abstract` constructor, die vervolgens allemaal `add_action('woocommerce_thankyou_' . $this->id, array($this, 'thankyou_page'));` aanroepen. 

Daardoor wordt op de 'thank you' pagina (na de betaling nu vijf keer de melding 'Bedankt voor je betaling') weergegeven. 

Ik denk dat je even moet checken of de actie al is toegevoegd met `has_action` alvorens hem toe te voegen. Ik vraag me sowieso af of er in de constructor een nieuwe instantie van `Mollie_WC_Gateway_DirectDebit` moet worden aangemaakt.